public/app/plugins/datasource/opentsdb/datasource.js

```diff
@@ -293,11 +294,12 @@ function (angular, _, dateMath) {
       var interpolatedTagValue;
       return _.map(metrics, function(metricData) {
         return _.findIndex(options.targets, function(target) {
-          return target.metric === metricData.metric &&
-            _.all(target.tags, function(tagV, tagK) {
-            interpolatedTagValue = templateSrv.replace(tagV, options.scopedVars);
-            return metricData.tags[tagK] === interpolatedTagValue || interpolatedTagValue === "*";
-          });
+          return target.metric === metricData.query.metric &&
+          _.all(target.tags, function(tagV, tagK) {
+              interpolatedTagValue = templateSrv.replace(tagV, options.scopedVars);
+              return metricData.query.tags[tagK] === "literal_or("+interpolatedTagValue+")" || interpolatedTagValue === "*";
```

Comment:
So, in 2.2.0, Opentsdb has deprecated tags and introduced filters of 6 types. One of the type is "literal_or" (equivalent to piped tag values).

For example, when we send a tag key/value pair as following to the Opentsdb 2.2.0:
```env:prod-1|prod-2|prod-3```
Opentsdb converts this tag key/value into filters and returns the following:
```literal_or(env:prod-1|prod-2|prod-3)```

So, now to match the query, we need to match the request values to response values.